genty/genty.py

```diff
@@ -62,18 +62,29 @@ def _expand_datasets(test_functions):
         `iterator` of `tuple` of (`unicode`, `function`)
     :return:
         Generator yielding a tuple of
-        (method_name, unbound function, dataset name, dataset)
+        - method_name      : Name of the test method
+        - unbound function : Unbound function that will be the test method.
+        - dataset name     : String representation of the given dataset
+        - dataset          : Tuple representing the args for a test
+        - param factory    : Function that returns params for the test method
     :rtype:
         `generator` of `tuple` of
-        (`unicode`, `function`, `unicode` or None, `tuple` or None)
+        (`unicode`, `function`, `unicode` or None, `tuple` or None, `function`)
     """
     for name, func in test_functions:
         datasets = getattr(func, 'genty_datasets', {})
         if datasets:
             for dataset_name, dataset in six.iteritems(datasets):
-                yield name, func, dataset_name, dataset
-        else:
-            yield name, func, None, None
+                yield name, func, dataset_name, dataset, None
+
+        deferred_datasets = getattr(func, 'genty_deferred_datasets', [])
+        if deferred_datasets:
```

Comment:
Several reasons
1) Python 2 doesn't have great support for nested generators. In Python 3.3 there's "yield from generator"
2) doctrings. The verbosity & repetitiveness of the docstrings saps my will to live :). I *love* the appropriate level of doctrings, but loath have to repeat things over and over.

I am able, however, to refactor in place such merging the 2 branches into a single looping construct